app/src/main/java/co/nano/nanowallet/ui/intro/IntroWelcomeFragment.java

```diff
@@ -72,14 +66,14 @@ public void onClickNewWallet(View view) {
                     Credentials credentials = realm.createObject(Credentials.class);
                     credentials.setSeed(NanoUtil.generateSeed());
                 });
-
-                sharedPreferencesUtil.setFromNewWallet(true);
-                ((WindowControl) getActivity()).getFragmentUtility().replace(
-                        IntroLegalFragment.newInstance(),
-                        FragmentUtility.Animation.ENTER_LEFT_EXIT_RIGHT,
-                        FragmentUtility.Animation.ENTER_RIGHT_EXIT_LEFT,
-                        IntroLegalFragment.TAG
-                );
+                // if this preference was saved successfully, launch the Legal Fragment
+                if (sharedPreferencesUtil.setFromNewWallet(true, false)) {
+                    showIntoLegalFragment();
+                }else{ // otherwise save preference in the background thread
+                    sharedPreferencesUtil.setFromNewWallet(true, true);
+                    showIntoLegalFragment();
```

Comment:
You should pull out showIntoLegalFragment() since you have it in both conditions. The formatting appears to be a bit off too on your else.

Reply:
Done.